src/ansys/fluent/core/streaming_services/events_streaming.py

```diff
@@ -225,3 +243,57 @@ def start(self, *args, **kwargs) -> None:
     def stop(self) -> None:
         """Stop streaming."""
         self._impl.stop()
+
+    def _register_solution_event_sync_callback(
+        self,
+        event_type: Literal[
+            SolverEvent.ITERATION_ENDED_SYNC, SolverEvent.TIMESTEP_ENDED_SYNC
+        ],
+        callback_id: str,
+        callback: Callable,
+    ):
+        unique_id = self._session.scheme_eval.scheme_eval(
```

Comment:
The unique_id is streamed through the auto-pause event which is checked within the client-side callback.